### ISSUE TITLE
[2] Logout does not work

### ISSUE THREAD

**Issue by calvinfroedge** [bitshares/bitshares-ui#1360]
User cannot logout of bitshares wallet without totally clearing all local storage. When I click logout I expect to be able to initiate a new session with a new username. This is not possible unless I clear all local settings via the developer console.

![update](https://user-images.githubusercontent.com/632938/37877241-8bbb1864-301d-11e8-9745-3c46c3e2d15a.gif)

**Comment by calvinfroedge** [comment]
The following sequence of actions seems to create a "totally unauthenticated state":

```
localStorage.clear()
iDB.root.deleteDatabase(true)
iDB.root.database_name = iDB.root.database_name.replace('db', 'v2') + '_default';
iDB.root.deleteDatabase(true)
```

This results in me even needing to reimport my wallet backup file to be able to login again. So I guess my question is this...I would expect a logout to produce a completely unauthenticated state such that no data about my account can be seen on the screen once I press the logout button. Am I wrong in my thinking?

@wmbutler @startailcoon

**Comment by wmbutler** [comment]
This is pretty serious. When did this get introduced?

**Comment by calvinfroedge** [comment]
Not sure, but I think it may have to do with the iDB / Web SQL, which I think is fairly new. I discovered this while exploring #1192

It may be helpful to show a confirmation before completely killing the session as doing the above (totally clearing local persistence) will result in needing to reimport the wallet backup.

**Comment by ahdigital** [comment]
> When did this get introduced?

Pretty sure has been the behaviour since at least January.

**Comment by calvinfroedge** [comment]
Anyway, I've essentially already identified the fix (how to totally kill the logout). What do you guys think about adding a confirmation on logout in local wallet mode to make sure people don't logout without realizing they need to download their backup file first? In cloud mode it doesn't really matter.

**Comment by svk31** [comment]
You're confusing cloud and wallet modes. In your licecap you're using wallet mode, there's no way to log out of a wallet, you only lock or unlock them. To switch wallets or create new ones, use the Wallet menu in the settings.

**Comment by wmbutler** [comment]
I see. ok, so yeah we have a twofold issue here. A user may want

1. Logout and retain their account name
2. Logout and clear the account name

We need UX to handle these cases gracefully and terminology to support it. @ahdigital

**Comment by calvinfroedge** [comment]
@svk31 Well this was certainly confusing me to say the least. I was trying to log out of wallet mode so I could log into a cloud wallet.

@wmbutler Posted another licecap, this is not locally, it's on live server. If I am in cloud wallet, I expect to click the logout button and refresh the page and see no reference to my first account. If I am in local wallet, I expect to actually be able to log out / clear it, not just lock / unlock. Locking / unlocking still leaves my data on the screen for anyone in the world to see.

![update](https://user-images.githubusercontent.com/632938/37878604-5246f788-3031-11e8-943b-3c815c530e29.gif)

**Comment by wmbutler** [comment]
@calvinfroedge It's a public blockchain. Anyone can navigate to your account to see all readonly info.

I use the local wallet exclusively and I would not be a fan of having my username lost on logout.

Regarding the cloud wallet, I can see a case for (a) keeping the account name stored or (b) clearing it so I could login to a different cloud wallet.

**Comment by calvinfroedge** [comment]
@wmbutler What if you want to switch between cloud wallet and local wallet? Is that something no one will ever do?

**Comment by wmbutler** [comment]
@calvinfroedge I'm sure there is a case for that, but I'm not sure what it looks like from a UX standpoint. It's dicy because I don't know if we are forced to clear out the localstorage items associated with the local wallet while using the cloud wallet.

It would be pretty cool if the user could switch to the cloud wallet, login and do what they needed to do then switch back to the local wallet without issue.

Please take a look at the UX for a more unified approach to this as well. Scroll to the very bottom for the various scenarios:

https://github.com/bitshares/bitshares-ui/issues/895

**Comment by ahdigital** [comment]
To these points:

@calvinfroedge 

>If I am in cloud wallet, I expect to click the logout button and refresh the page and see no reference to my first account.

I agree.

@wmbutler 

> A user may want
> - Logout and retain their account name
> - Logout and clear the account name

I agree.

So, perhaps it is better to clear account name etc. on logout and add an option 'Remember me' to the login process.

**Comment by wmbutler** [comment]
As long as "remember me" is default, we may have a winner here. But it's still dangerous to "forget" the user on a local wallet logout because without a backup, the account will be lost.

**Comment by ahdigital** [comment]
> But it's still dangerous to "forget" the user on a local wallet logout because without a backup, the account will be lost.

Piecing our efforts together, the user gets a prompt to backup upon account creation, a warning in the footer and per #895 a warning on login.

Arguably, we could make the footer warning more prominent like the one I've just proposed in #1190.

Or, as an additional safety net we could display an alert modal on click of the logout button if the user meets this criteria:

1. Using a local wallet (key file)
2. No backup detected

**Comment by svk31** [comment]
You should never call `iDB.root.deleteDatabase(true)` or attempt to delete the database, the only way to do that should be via the settings where you already have the option to delete a wallet.

Switching between modes is also done in the Settings. In the case where you're already using a wallet with accounts in it and switch to cloud mode, the wallet accounts will still show since the loading of databases is so intertwined in how the GUI works I found it easier to just leave it like that. One possible solution to this is to create a dummy wallet with no accounts in it, which can be loaded whenever cloud mode is activated.

**Comment by wmbutler** [comment]
I think there is value in being able to switch to the cloud wallet and retain the local wallet credentials. The dummy wallet idea may have merit.

@svk31 if the user were to switch back to the local wallet, are you suggesting that they would not need to re-import keys? The local wallet would be in place just as it had before they performed their cloud wallet operations?

**Comment by svk31** [comment]
I don't think we should ever force the user to reimport keys. The dummy wallet I explained above would simply be a way to have an empty wallet without any keys in it, which we could use to present a "clean slate" when you switch to cloud mode.

**Comment by ahdigital** [comment]
The idea for a dummy account makes sense to me. I don't see why not, but would it play nice with the proposal to consolodate login, #895? which I've summarized in my most recent post.

**Comment by wmbutler** [comment]
What if we push this to 180501?
